examples/cpp/01_gemm/util.hpp

```diff
@@ -1,35 +1,59 @@
 #pragma once
 
+#include "util/cuda_timer.hpp"
+
 #include <thrust/device_vector.h>
 #include <thrust/host_vector.h>
 
 #include <cfloat>
 
-float rand_float(float a = 1e-3, float b = 1) {
+float rand_float(float a = 1e-4, float b = 1e-2) {
     float random = ((float)rand()) / (float)RAND_MAX;
     float diff = b - a;
     float r = random * diff;
     return a + r;
 }
 
-// In this implementation, A and C are interpreted as being laid out in
-// row-major, and B is interpreted as being laid out in column-major.
-void naive_gemm(int kM, int kN, int kK,  //
-                const __half* A, const __half* B, float* C) {
-    for (int i = 0; i < kM; ++i) {
-        for (int j = 0; j < kN; ++j) {
-            float s = 0.;
-            for (int k = 0; k < kK; ++k) {
-                s += __half2float(A[i * kK + k]) * __half2float(B[k + kK * j]);
-            }
-            C[i * kN + j] = s;
+float cublas_hgemm(int64_t kM, int64_t kN, int64_t kK,  // problem shape
+                   const __half* A, const __half* B, __half* C,
+                   bool timeit = false) {
+    cublasHandle_t handle;
+    cublasCreate(&handle);
+
+    __half alf = static_cast<__half>(1.);
+    __half bet = static_cast<__half>(0.);
+
+    float elapsed = 0.;
+
+    if (timeit) {
+        for (int i = 0; i < 5; ++i) {
+            cublasHgemm(handle, CUBLAS_OP_T /* transb*/, CUBLAS_OP_N, kN, kM,
+                        kK, &alf, B, kK, A, kK, &bet, C, kN);
         }
+        cudaDeviceSynchronize();
+
+        int iters = 20;
```

Comment:
Do we need to include `iters` in the parameter list?

Reply:
fixed.